### PR TITLE
docs

### PR DESCRIPTION
**Pull request by brainrake**
see #37

- [x] `nix build .#docs` static site generator
- [x] add readme to docs site
- [x] auto-generated api docs
- [x] generate navigation index
- [x] fix up links in generated documentation
- [x] docs commands in devenv
- [x] publish gh-pages
- [x] work around godot and gdscript-docs-maker bugs
- [x] fix pdf download

Generating docs for `provider.gd` crashes so it is removed, pending a fix in godot engine.
https://github.com/godotengine/godot/issues/81598

### CODE REVIEW

File: docs/default.nix [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        # workaround for bug
+        # https://github.com/GDQuest/gdscript-docs-maker/pull/97
+        sed -i 's/_init(/_initialize(/g' ReferenceCollectorCLIGd4.gd
+
+        # set input path
+        sed -i 's/\"res\:\/\/\"/\"res:\/\/addons\/@mlabs-haskell\/godot-cardano\"/g' ReferenceCollectorCLIGd4.gd
+        cat ReferenceCollectorCLIGd4.gd
+
+        # work around crash in provider.gd
+        # https://github.com/godotengine/godot/issues/81598
```
Comment (brainrake) [inline docs/default.nix:20]: :(

**Review by rmgaray — CHANGES_REQUESTED** [review]
I can't say I understand yet how all of this comes together, but I did find an issue while testing out the PR.

I've tried adding a description to a file as explained [here](https://docs.godotengine.org/en/latest/tutorials/scripting/gdscript/gdscript_documentation_comments.html#documenting-a-script)  but the generator fails to include the brief description and also duplicates the description itself.

This is what I added at the beginning of `single_address_wallet.gd`:

```
extends RefCounted
## A brief description of the class.
##
## A longer description of what the class does and how to initialize it.

class_name SingleAddressWallet
``` 

But the generator creates this:

imagen

I noticed this because we need to revise our docs to properly follow the GDScript doc comment system (all of our descriptions are broken I believe).

**Comment by brainrake** [comment]
Strangely, the built-in documentation system in godot editor (press F1, type class name) shows the description correctly. So it's probably either a bug in godot language server related code or the reference export script.

**Review by rmgaray — APPROVED** [review]
I like how the docs look and I have nothing to add to the implementation (we will have to do with these workarounds until some of the issues are fixed upstream). Good work, @brainrake !

**Pull request merged**